src/main/java/com/google/cloud/spanner/jdbc/JdbcArray.java

```diff
@@ -167,6 +186,34 @@ public ResultSet getResultSet(long startIndex, int count) throws SQLException {
     JdbcPreconditions.checkArgument(startIndex >= 1L, "Start index must be >= 1");
     JdbcPreconditions.checkArgument(count >= 0, "Count must be >= 0");
     checkFree();
+    Type spannerType = type.getSpannerType();
```

Comment:
`type` cannot be null here because to create a `JdbcArray` using [createArray](https://github.com/googleapis/java-spanner-jdbc/blob/8f75d26c18844617e275feb952db415a067a91d2/src/main/java/com/google/cloud/spanner/jdbc/JdbcArray.java#L57) we need a valid type as input. An invalid type throws an [exception](https://github.com/googleapis/java-spanner-jdbc/blob/8f75d26c18844617e275feb952db415a067a91d2/src/main/java/com/google/cloud/spanner/jdbc/JdbcArray.java#L63).

Also, `type.getSpannerType()` is already used in the [code](https://github.com/googleapis/java-spanner-jdbc/blob/8f75d26c18844617e275feb952db415a067a91d2/src/main/java/com/google/cloud/spanner/jdbc/JdbcArray.java#L230) today without null check.